src/main/java/io/confluent/copycat/hdfs/partitioner/Partitioner.java

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright 2015 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ **/
+
+package io.confluent.copycat.hdfs.partitioner;
+
+import org.apache.kafka.copycat.sink.SinkRecord;
+
+import java.util.Map;
+
+/**
+ * Partition incoming records, and generates directories and file names in which to store the
+ * incoming records.
+ */
+public interface Partitioner {
+  void configure(Map<String, Object> config);
+  String encodePartition(SinkRecord sinkRecord);
+  String encodePartition(Object partitionKey);
+  String generatePartitionedPath(String topic, String encodedPartition);
```

Comment:
Can you explain why we need both `encodePartition(SinkRecord)` and `generatePartitionedPath`? It seems like if we generated the directory name we wanted to place the data under, that should be enough. Both are strings anyway so they seem equivalent.

More generally, the issue I mentioned about daylight savings time -- I'm wondering if it would be more reliable to have joda-time handle extracting just the date info and formatting it without us having to handle dividing the timeline up into even pieces. In particular, I'm worried that for something like the hourly partitioner, we currently might be generating the same directory name for two different blocks of time when daylight savings time hits (when the clock rolls back, we see the hour 2am - 3am 2x, but the unix timestamps are different). I'm honestly now sure how people would prefer for this to be handled....